### DISCUSSION TITLE
Curl with Openssl for Windows

### DISCUSSION THREAD

**Discussion by artenax** [curl/curl#15448] [Q&A]
I want to compile Curl with Openssl for Windows, but I'm using Linux. How do I compile Curl with Openssl for Windows?
Preferably mingw, not msvc.

**Comment by bagder** [comment]
Here's what everything curl says about it: https://everything.curl.dev/build/windows.html

(Oops, it does not mention cross-compiles, it mentions native builds.)

**Comment by artenax** [comment]
I successfully cross compiled curl with schannel (configure + mingw), but curl never saw openssl.

**Comment by artenax** [comment]
Well, cmake with -DCMAKE_TOOLCHAIN_FILE sees openssl (no ECH, of course), but there are other problems there. Windows is a very problematic OS.
